crates/next-transform-strip-page-exports/src/lib.rs

```diff
@@ -0,0 +1,897 @@
+                    id.0,
+                    id.1,
+                    this.in_removed_item
+                );
+
+                d.visit_children_with(this);
+            });
+        });
+    }
+}
+
+impl Visit for Analyzer<'_> {
+    // This is important for reducing binary sizes.
+    noop_visit_type!();
+
+    fn visit_binding_ident(&mut self, i: &BindingIdent) {
+        if !self.in_lhs_of_var || self.in_removed_item {
+            self.add_ref(i.id.to_id());
+        }
+    }
+
+    fn visit_named_export(&mut self, n: &NamedExport) {
+        for specifier in &n.specifiers {
+            if let Some(ExportTypeResult {
+                exported_ident,
+                local_ident,
+                export_type,
+            }) = ExportType::from_specifier(specifier)
+            {
+                self.state
+                    .encounter_export(exported_ident, local_ident, export_type);
+
+                if let Some(local_ident) = local_ident {
+                    if self.state.should_retain_export_type(export_type) {
+                        self.add_ref(local_ident.to_id());
+                    }
+                }
+            }
+        }
+    }
+
+    fn visit_export_decl(&mut self, s: &ExportDecl) {
+        match &s.decl {
+            Decl::Var(d) => {
+                for decl in &d.decls {
+                    if let Pat::Ident(ident) = &decl.name {
+                        if let Some(export_type) = ExportType::from_ident(ident) {
+                            self.state.encounter_export(ident, None, export_type);
+
+                            let retain = self.state.should_retain_export_type(export_type);
+
+                            if retain {
+                                self.add_ref(ident.to_id());
+                            }
+
+                            self.within_removed_item(!retain, |this| {
+                                decl.visit_with(this);
+                            });
+                        } else {
+                            // Always preserve declarations of unknown exports.
+                            self.add_ref(ident.to_id());
+
+                            decl.visit_with(self)
+                        }
+                    } else {
+                        decl.visit_with(self)
+                    }
+                }
+            }
+            Decl::Fn(decl) => {
+                let ident = &decl.ident;
+                if let Some(export_type) = ExportType::from_ident(ident) {
+                    self.state.encounter_export(ident, None, export_type);
+
+                    let retain = self.state.should_retain_export_type(export_type);
+
+                    if retain {
+                        self.add_ref(ident.to_id());
+                    }
+
+                    self.within_removed_item(!retain, |this| {
+                        decl.visit_with(this);
+                    });
+                }
```

Comment:
I'm not sure if this is intended, but I think we need `else` + `decl.visit_with(this);`

Reply:
I'm pretty sure we do as well. Let me write a test case for this.